File: 🟢 Easy/989 - Add to Array-Form of Integer/_989AddtoArray-FormofInteger.java
class Solution {
    public List<Integer> addToArrayForm(int[] num, int k) {
        List<Integer> res = new ArrayList<>();
        int i = num.length - 1;
        while (i >= 0 || k > 0) {
            if (i >= 0) {
                k += num[i];
            }
            res.add(k % 10);
            k /= 10;
            i--;
        }
        Collections.reverse(res);
        return res;
    }
}R

File: 🟢 Easy/989 - Add to Array-Form of Integer/_989AddtoArray-FormofInteger.py
class Solution:
    def addToArrayForm(self, num: List[int], k: int) -> List[int]:
        res = []
        i = len(num) - 1
        while i >= 0 or k > 0:
            if i >= 0:
                k += num[i]

            res.append(k % 10)
            k //= 10
            i -= 1

        res.reverse()
        return res
